test/functional/wallet_assumeutxo.py

```diff
@@ -160,7 +171,7 @@ def run_test(self):
         self.wait_until(lambda: len(n1.getchainstates()['chainstates']) == 1)
 
         self.log.info("Ensuring wallet can be restored from backup")
-        n1.restorewallet("w", "backup_w.dat")
+        n1.restorewallet("w2", "backup_w2.dat")
```

Comment:
I have amended the log message and I have added basic coverage for the balances of the wallet. That required some further changes to the test as a whole since  there were no balances before due to the use of miniwallet, so I have done this in a separate commit.